github_checkpatch.py

```diff
@@ -366,13 +368,12 @@ def post_review(self, review_input):
                                                     self.repo)
                         num_annotations += 1
                     except KeyError:
-                        if path in review_input['files']:
-                            # not a line modified in the patch, add it to the
-                            # general message
-                            extra_review_comment += "[{0}:{1}](https://github.com/{4}" \
-                                                    "/{5}/blob/{3}/{0}#L{1}): {2}".format(
-                                                        path, comment['line'], comment['message'],
-                                                        commit.sha, self.project, self.repo)
+                        # not a line modified in the patch, add it to the
+                        # general message
+                        extra_review_comment += "[{0}:{1}](https://github.com/{4}" \
```

Comment:
Why do we comment if the line isn't in the patch but not if the file isn't in the patch?

Reply:
We could change this one day.  But currently I think this is just due to the potential land-slide of errors that would be reported about, if we commented on every error found and the relevance to the patch is about the worst it could be.

Limiting it to the files that were modified in the patch limits the number of complaints to being the most relevant: for lines modified by the patch, and secondarily relevant: to the files the patch is modifying.

I guess the hope is that while a developer is never going to fix all of the errors across the code base (without it being a specific project to do that) [s]he might feel inclined to fix the ones in the files [s]he is modifying if they are pointed out to him/her.

In any case, errors not in lines modified are not supposed to count towards whether the checkpatch marks a failure.